File: pref/locale/locale_android.java
package org.gioui.x.pref.locale;

import android.content.res.Resources;
import android.os.Build;
import java.util.Locale;

public class locale_android {
	public static String getLanguage() {
	    if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.LOLLIPOP) {
	        return "";
	    }

        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.N) {
           return Resources.getSystem().getConfiguration().locale.toLanguageTag();
        }

        return Resources.getSystem().getConfiguration().getLocales().get(0).toLanguageTag();
	}
}

File: pref/locale/locale_android.go
package locale

import (
	"gioui.org/x/pref/internal/xjni"
)

//go:generate javac -source 8 -target 8 -bootclasspath $ANDROID_HOME/platforms/android-29/android.jar -d $TEMP/x_locale/classes locale_android.java
//go:generate jar cf locale_android.jar -C $TEMP/x_locale/classes .

var (
	_Lib = "org/gioui/x/pref/locale/locale_android"
)

func getLanguage() string {
	lang, err := xjni.DoString(_Lib, "getLanguage", "()Ljava/lang/String;")
	if err != nil {
		return ""
	}

	return lang
}
